### DISCUSSION TITLE
Improving accuracy: Vue readings are 2-5% below what is reported by utility

### DISCUSSION THREAD

**Discussion by meetc0** [emporia-vue-local/esphome#312] [Q&A]
My utility has support for Green Button, and can output a XML or CSV with usage data in 15 minute increments, measured in watt-hours (not kilowatt). Home Assistant (by default) only shows 60 minute increments on the energy dashboard, but that's a different topic. I am getting an inconsistent variation between what the utility reports, and what the Vue collects. I have already completed voltage calibration on both phases, and the reported Vue voltage is very accurate compared to that of a multimeter. 

Here is a list of total daily usage for July this year. Ignore the discrepancy on July 11 and 14, that was due to downtime of HomeAssistant and/or the Vue
![dailykwh](https://github.com/user-attachments/assets/7ff1e89c-dedf-4d57-aa94-6a6d2c375ded)

And by hour energy for July 1
![0701kwh](https://github.com/user-attachments/assets/c9aa39d9-403c-40a8-ac58-a2f6e99b1457)

Other than the voltage calibration, how else can accuracy be improved for the data collection by the Vue?

**Comment by garylankford8** [comment]
I could be wrong, but I think this is what I'd try...

Throw out the outliers in your observations and average the percentages (looks like ~2.5%).

In the CT section, where a multiplication factor of 2 is usually applied to 240v circuits, apply a factor of 1.025 to 120v circuits, and 2.050 to 240v circuits.

**Reply by meetc0** [reply]
Can a multiplier be added to inputs A and B for the mains?

**Reply by garylankford8** [reply]
Not in the same way, and I don't know how to do it...

Could probably be done with a lambda calculation; something I need to learn!

Sorry I'm not more help.

**Comment by flaviut** [comment]
You could update the `calibration:` field you've already updated so that the power comes out to be correct, even if the voltage isn't quite right.  If you want the voltage to be correct too then, you could add a multiplier filter on the voltage configs.

But in general I'd say that 2.5% percent is pretty good! I use mine for automations & identifying power hogs, and for that a few percents don't make much of a difference.

**Reply by meetc0** [reply]
Can the multiplier: 2 on the 16 inputs take a float value instead of an integer? I would be able to add a calibration on each individual ct clamp if this can be done.

**Reply by flaviut** [reply]
Absolutely. I use ~1.7 because of my 3-phase service.

**Comment by cbrodine** [comment]
How far away from your meter is your panel? 2% would be a reasonable amount of line loss to be experiencing if you have enough distance between the two metering points.

**Reply by meetc0** [reply]
Under 2m. voltage drop isn't the issue.

**Comment by gekkehenkie11** [comment]
If I remember correctly, Emporia stated that the measurements deviate max 3%. So if you're within 3% then everything is normal.
